packages/xo-web/src/xo-app/hub/resource-form.js

```diff
@@ -3,38 +3,65 @@ import _ from 'intl'
 import decorate from 'apply-decorators'
 import Icon from 'icon'
 import React from 'react'
+import SingleLineRow from 'single-line-row'
 import Tooltip from 'tooltip'
-import { Container } from 'grid'
-import { SelectPool } from 'select-objects'
-import { error } from 'notification'
+import { Container, Col } from 'grid'
+import { find, first } from 'lodash'
+import { SelectPool, SelectSr } from 'select-objects'
+import { isSrWritable } from 'xo'
 import { injectState, provideState } from 'reaclette'
 
 export default decorate([
   provideState({
     initialState: ({ multi }) => ({
       pools: multi ? [] : undefined,
+      mapPoolsSrs: {},
     }),
     effects: {
       onChangePool(__, pools) {
-        const noDefaultSr = Array.isArray(pools)
-          ? pools.some(pool => pool.default_SR === undefined)
-          : pools.default_SR === undefined
-        if (noDefaultSr) {
-          error(_('hubSrErrorTitle'), _('noDefaultSr'))
-        } else {
-          this.props.onChange({
-            pools,
-            pool: pools,
-          })
-          return {
-            pools,
+        this.props.onChange({
+          pools,
+          mapPoolsSrs: this.state.mapPoolsSrs,
+        })
+        const _mapPoolsSrs = {}
+        if (Array.isArray(pools)) {
```

Comment:
I think it would be more readable if you did this instead:
```js
pools = Array.isArray(pools) ? pools : [pools]
```

Reply:
It's better, but unfortunately it will not work because `Select` does'nt accept `array` for signle selection (`multi="false"`)

Reply:
Ok, I didn't see you were re-using `pools` later, but then:
```js
const _pools = Array.isArray(pools) ? pools : [pools]
```